qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -221,9 +255,79 @@ def get_fidelity(
         circuit: QuantumCircuit,
         backend: Optional[Union[Backend, QuantumInstance]] = None,
         expectation: Optional[ExpectationBase] = None,
+        *,
+        sampler: Optional[BaseSampler] = None,
     ) -> Callable[[np.ndarray, np.ndarray], float]:
         r"""Get a function to compute the fidelity of ``circuit`` with itself.
 
+        .. note::
+
+            Using this function with a backend and expectation converter is pending deprecation,
+            instead pass a Qiskit Primitive sampler, such as :class:`~.Sampler`.
+
+        Let ``circuit`` be a parameterized quantum circuit performing the operation
+        :math:`U(\theta)` given a set of parameters :math:`\theta`. Then this method returns
+        a function to evaluate
+
+        .. math::
+
+            F(\theta, \phi) = \big|\langle 0 | U^\dagger(\theta) U(\phi) |0\rangle  \big|^2.
+
+        The output of this function can be used as input for the ``fidelity`` to the
+        :class:~`qiskit.algorithms.optimizers.QNSPSA` optimizer.
+
+        Args:
+            circuit: The circuit preparing the parameterized ansatz.
+            backend: *Pending deprecation.* A backend of quantum instance to evaluate the circuits.
+                If None, plain matrix multiplication will be used.
+            expectation: *Pending deprecation.* An expectation converter to specify how the expected
+                value is computed. If a shot-based readout is used this should be set to
+                ``PauliExpectation``.
+            sampler: A sampler primitive to sample from a quantum state.
+
+        Returns:
+            A handle to the function :math:`F`.
+
+        """
+        # allow passing sampler by position
```

Comment:
I guess you are looking to drop the keyword requirement going forwards. At present the interface does not give an cluse that backend parameter can take a Sampler too.

Reply:
Oh good point! I added a comment in the docstring and the example in the class doc now also passes it positionally.